core/src/data/value/selector.rs

```diff
@@ -45,4 +45,26 @@ impl Value {
 
         Ok(value)
     }
+
+    pub fn selector_by_index(&self, selector: &[Value]) -> Result<Value> {
+        let str_keys = selector.iter().map(String::from).collect::<Vec<_>>();
+
+        let value = str_keys.iter().try_fold(self, |selectable, key| {
+            selectable.get_value_from_compound_type(key)
+        })?;
+
+        Ok(value.clone())
+    }
```

Comment:
```suggestion
    pub fn selector_by_index(&self, selector: &[Value]) -> Result<Value> {
        selector
            .iter()
            .map(String::from)
            .try_fold(self, |selectable, key| {
                selectable.get_value_from_compound_type(&key)
            })
            .map(Clone::clone)
    }
```

It looks like you don't need to consume the iterator as a `collect` in the middle.